### ISSUE TITLE
K.gradients return None over a layer.output or layer.input (except model input)

### ISSUE THREAD

**Issue by tchaton** [keras-team/keras#9992]
Hello,

I am trying to get activations and their gradient for each trainable layer output.
However, the function returns me a None tensor.

model.optimizer.get_gradients(model.total_loss, layer.output) 
or 
K.gradients(model.total_loss, layer.output) 

Can you help me please ?

**Comment by tchaton** [comment]
It works also with the dense layer
[<tf.Tensor 'gradients_44/vgg16_1/block1_conv1/convolution_grad/Conv2DBackpropInput:0' shape=(?, ?, ?, 3) dtype=float32>, None, None, None, None, None, None, None, None, None, None, None, None, <tf.Tensor 'gradients_44/dense_1_1/MatMul_grad/MatMul:0' shape=(?, 512) dtype=float32>]
But every from the CNN gets a None tensor

**Comment by tchaton** [comment]
Ok I was able to access the gradients but this error should be corrected. 
Here is my solution 
I have two models with the same layers but one as some wrapped into a model.
First, ['input_2', 'vgg16', 'dense_2']
Two, ['input_1',
 'block1_conv1',
 'block1_conv2',
 'block1_pool',
 'block2_conv1',
 'block2_conv2',
 'block2_pool',
 'block3_conv1',
 'block3_conv2',
 'block3_conv3',
 'block3_pool',
 'block4_conv1',
 'block4_conv2',
 'block4_conv3',
 'block4_pool',
 'block5_conv1',
 'block5_conv2',
 'block5_conv3',
 'block5_pool',
 'global_average_pooling2d_2',
 'dense_2']
I train with the wrapped, save the weight then put them into the linear one using the followings functions
from keras.models import Model
def get_layer_rec(model, layer_name):
	out_layer = None
	for layer in model.layers: #order_layers(model.layers):
		if isinstance(layer, Model):
			out_layer = get_layer_rec(layer, layer_name)
		else:
			if layer.name == layer_name:
				out_layer = layer
	return out_layer
def insertSymbolic(model, modelSymbo):
    layers_errors = []
    for layer in modelSymbo.layers[1:]:
        try:
            name = layer.name
            layer_ref = get_layer_rec(model, name)
            layer.set_weights(layer_ref.get_weights())
        except:
            layers_errors.append(name)
    return modelSymbo, layers_errors

**Comment by skeydan** [comment]
This seems to be the same thing

https://github.com/rstudio/keras/issues/472#issuecomment-405234380

would anyone have any news regarding this issue?
